dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/HibernateTrackedEntityInstanceStoreTest.java

```diff
@@ -0,0 +1,156 @@
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.tracker;
+
+import static org.hisp.dhis.tracker.Assertions.assertNoErrors;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import org.hisp.dhis.common.DimensionalItemObject;
+import org.hisp.dhis.common.QueryItem;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.trackedentity.TrackedEntityInstanceQueryParams;
+import org.hisp.dhis.trackedentity.hibernate.HibernateTrackedEntityInstanceStore;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.webapi.controller.event.mapper.OrderParam;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+class HibernateTrackedEntityInstanceStoreTest extends TrackerTest
+{
+
+    @Autowired
+    private TrackerImportService trackerImportService;
+
+    @Autowired
+    private HibernateTrackedEntityInstanceStore teiStore;
+
+    private TrackedEntityInstanceQueryParams params;
+
+    @Override
+    protected void initTest()
+        throws IOException
+    {
+        setUpMetadata( "tracker/simple_metadata.json" );
+        User userA = userService.getUser( "M5zQapPyTZI" );
+        assertNoErrors(
+            trackerImportService
+                .importTracker( fromJson( "tracker/tracked_entity_basic_data_for_ordering.json", userA.getUid() ) ) );
+        manager.flush();
+
+        params = new TrackedEntityInstanceQueryParams();
+    }
+
+    @Test
+    void whenOrderingByCreatedAtThenEntitiesAreSortedById()
+    {
+        params.setOrders( List.of( new OrderParam( "createdAt", OrderParam.SortDirection.ASC ) ) );
+
+        List<Long> list = teiStore.getTrackedEntityInstanceIds( params );
+
+        assertEquals( "[1, 2, 3, 4]", list.toString() );
+    }
+
```

Comment:
We should make assertEquals comparing lists instead of strings. For this, we can create `TrackedEntityInstance` in a specific order before each test or in the test so we have objects to compare